web/src/portico/signup.ts

```diff
@@ -503,7 +503,15 @@ $(() => {
             if (current_countdown > 0) {
                 $countdown_elt.text((current_countdown - 1).toString());
             } else {
-                window.location.href = $("a#deactivated-org-auto-redirect").attr("href")!;
+                let new_org_url = $("a#deactivated-org-auto-redirect").attr("href")!;
+                const url_hash = window.location.hash;
+                if (url_hash.startsWith("#")) {
+                    // Ensure we don't double-add hashes and handle query parameters properly
+                    const url = new URL(new_org_url);
+                    url.hash = url_hash;
+                    new_org_url = url.toString();
+                }
+                window.location.href = new_org_url;
```

Comment:
I'm pretty sure that doing this fix upon redirect leaves the link in the page incorrect, in case the user is impatient and clicks it before the 5s countdown.

We should fix the link on page load, so that it's correct in case it's clicked on.

Reply:
Yeah, added a commit for that.